main/src/ca/uwaterloo/flix/language/phase/EntryPoints.scala

```diff
@@ -584,11 +568,7 @@ object EntryPoints {
             duplicatedHandlerErrors += EntryPointError.DuplicateDefaultHandler(handledSym, loc2, loc1)
         }
       }
-      val defsWithDefaultHandlers = ParOps.parMapValues(root.defs)(
-        defn =>
-          if (isEntryPoint(defn)(root)) wrapDefWithDefaultHandlers(defn, validHandlers, root) else defn
-      )
-      (root.copy(defs = defsWithDefaultHandlers), errors.toList.flatMap(_.toList) ++ duplicatedHandlerErrors)
+      (root.copy(defaultHandlers = validHandlers.toList), errors.toList.flatMap(_.toList) ++ duplicatedHandlerErrors)
```

Comment:
Perhaps the return type would be more clear if it was a list of DefaultHandlers?